lib/galaxy/datatypes/sniff.py

```diff
@@ -487,7 +487,7 @@ def handle_uploaded_dataset_file( filename, datatypes_registry, ext='auto', is_m
         ext = guess_ext( filename, sniff_order=datatypes_registry.sniff_order, is_multi_byte=is_multi_byte )
 
     if check_binary( filename ):
-        if not Binary.is_ext_unsniffable(ext) and not datatypes_registry.get_datatype_by_extension( ext ).sniff( filename ):
+        if not datatypes.binary.Binary.is_ext_unsniffable(ext) and not datatypes_registry.get_datatype_by_extension( ext ).sniff( filename ):
```

Comment:
No, I don't think you should change sniff.py.
I think you need very few things from the Sam class.
In fact I think you only need the column names, so you could just initialize them in the Bam class.